tribe-ext-template-outliner.php

```diff
@@ -89,45 +104,44 @@ private function get_settings() {
 
 		/**
 		 * Extension initialization and hooks.
+		 *
+		 * @since 1.0.0
 		 */
 		public function init() {
-			// Load plugin textdomain
-			// Don't forget to generate the 'languages/tribe-ext-template-outliner.pot' file
+			// Load plugin textdomain.
 			load_plugin_textdomain( 'tribe-ext-template-outliner', false, basename( dirname( __FILE__ ) ) . '/languages/' );
 
 			if ( ! $this->php_version_check() ) {
 				return;
 			}
 
-			// Don't run in the admin or anything silly like that!
-			if (
-				is_admin()
-				|| wp_doing_ajax()
-			) {
-				return;
-			}
+			$this->class_loader();
+
+			$this->get_settings();
 
-			// Don't show to the general public - in case someone actually uses it on a live site!
+			// Only show to the roles allowed in the settings.
 			$user = wp_get_current_user();
-			if ( ! in_array( 'administrator', (array) $user->roles ) ) {
+			$allowed = $this->settings->get_option( 'roles', [] );
+			
+			$is_allowed = array_intersect( $allowed, (array) $user->roles );
+
+			if ( empty( $is_allowed )  ) {
 				return;
 			}
 
-			$this->class_loader();
-
-			// No settings for now!
-			//$this->get_settings();
-
 			// Insert filter and action hooks here
-			add_action( 'wp_head', [ $this, 'tribe_ext_template_outliner_styles' ], 100 );
-			add_action( 'wp_print_footer_scripts', [ $this, 'tribe_ext_template_outliner_scripts' ], 100 );
+			add_action( 'init', [ $this, 'tribe_ext_template_outliner_styles' ], 100 );
+			add_action( 'init', [ $this, 'tribe_ext_template_outliner_scripts' ], 100 );
+			add_action( 'wp_print_footer_scripts', [ $this, 'tribe_ext_template_outliner_panel' ], 1 );
```

Comment:
unsure if tribe_asset() has support for wp_print_footer_scripts but might be worth looking into

Reply:
That's not a script - the function appends the panel to the body.